equinix/helpers_device.go

```diff
@@ -122,6 +166,21 @@ func getPorts(ps []packngo.Port) []map[string]interface{} {
 	return ret
 }
 
+func getMetalGoPorts(ps []metalv1.Port) []map[string]interface{} {
+	ret := make([]map[string]interface{}, 0, 1)
+	for _, p := range ps {
+		port := map[string]interface{}{
+			"name":   p.GetName(),
+			"id":     p.GetId(),
+			"type":   p.GetType(),
+			"mac":    p.Data.GetMac(),
```

Comment:
Data is `*PortData`, it could technically be null.
`p.GetData.GetMac()` would be nil-safe, at the risk of returning "" for mac and false for bonded if Data was not included in the response (or testing Port object). 

These zero values seem like they would be appropriate if `data` was not included in the response, for whatever reason.

Reply:
`.GetMac()` and `.GetBonded()` have pointer receivers, so they cannot be chained on `GetData()`.  We'd have to do `GetData().Mac` and `GetData().Bonded`, which are `*string` and `*bool` respectively, so we'll need to break these calls out of the map declaration so we can check for nil and use an appropriate fallback.

Reply:
Or...actually, I think we don't need a fallback, and `p.Data.Get<Blah>()` is doing what we want it to, because if `Data` is a nil `*PortData`, then e.g. [`p.Data.GetMac()` will return `""`](https://github.com/equinix-labs/metal-go/blob/main/metal/v1/model_port_data.go#L51).